### DISCUSSION TITLE
Forward auth in Kong/Azure Multi tenant support

### DISCUSSION THREAD

**Discussion by JonasKs** [Kong/kong#7779] [Ideas and feature requests]
Hi,  
A few short questions about Kong that I haven't really found clear answers for yet. 

Traefik has [ForwardAuth](https://doc.traefik.io/traefik/v2.0/middlewares/forwardauth/), which essentially is auth through a micro service. Is there anything similar in Kong? 

If not, is there Azure Multi tenant support, which allow for specific issuers only to be allowed? Or to have such custom logic ourselves? 

The third option is to write a custom plugin it seems? 

Thanks in advance 😊

**Comment by dndx** [comment]
Seems pretty similar to http://nginx.org/en/docs/http/ngx_http_auth_request_module.html, although Kong does not currently have this functionality built-in. Pinging @PaulFischer-Kong for some awareness.
